crates/kernel/testsuite/moot_suite.rs

```diff
@@ -0,0 +1,155 @@
+//! Moot is a simple text-based test format for testing the kernel.
+//!
+//! Example test:
+//!
+//! # This is a comment.
+//! ; return 42;
+//! 42
+//!
+//! # Empty lines are ignored
+//!
+//! # Both thrown and returned errors can be matched with a simple error value
+//! ; eval();
+//! E_ARGS
+//!
+//! # Multi-line commands: continuation with `>`.
+//! ; return 1 + 2 +
+//! > 3;
+//! 6
+//!
+
+mod common;
+use std::{
+    fs::File,
+    io::{BufRead, BufReader},
+    path::Path,
+    sync::Arc,
+};
+
+use common::create_db;
+use eyre::Context;
+use moor_db::odb::RelBoxWorldState;
+use moor_values::var::v_none;
+use pretty_assertions::assert_eq;
+
+use crate::common::WIZARD;
+
+#[derive(Debug, Default)]
+enum MootState {
+    #[default]
+    Ready,
+    ReadingCommand {
+        line_no: usize,
+        command: String,
+    },
+    ReadingExpectation {
+        line_no: usize,
+        command: String,
+        expectation: String,
+    },
+}
+impl MootState {
+    // Could implement this with `nom` I guess, but this seems simple enough, and it's probably easier to read.
```

Comment:
yeah the other option would be `pest` since we already have that as a dep, but that's overkill.